### ISSUE TITLE
req parameter being discarded

### ISSUE THREAD

**Issue by developius** [getsentry/raven-node#333]
**Do you want to request a *feature* or report a *bug*?**
Bug

**Has someone had this problem before?**

Not as far as I can see.

**What is the current behavior?**

Config:
```js
Raven.config(DSN, {
  release: pkg.version,
  environment: process.env.NODE_ENV,
  sendTimeout: 60,
  parseUser: req => {
    // custom user parsing logic
    return {
      email: req.user.email,
      username: req.user.username,
      id: req.user.id
    }
  }
}).install()

utilities.Raven.captureMessage('Slow route', { req, level: 'warning', extra: { duration } }, (err, eventId) => {
  console.log(eventId)
})
```

The UI is reporting `There was 1 error encountered while processing this event` and the expanded error is `Discarded invalid parameter 'req'` with this for context:

```json
{
  "name": "req"
}
```

`req` is correctly populated (I checked via the `dataCallback` option) so I'm not sure why Sentry is discarding it?

**What is the expected behavior?**

I would expect this code to work correctly, as it appears to be the same as in the docs. This is the Node client for Raven SDK (2.1.0 & 2.0.2) and the hosted version of Sentry (8.18.0.dev0 apparently)

**Comment by developius** [comment]
I just encountered an issue which may help - Sentry reported an error in this code:

`Cannot read property 'email' of undefined`

```js
Raven.config(DSN, {
  release: exports.revision,
  environment: process.env.NODE_ENV,
  sendTimeout: 60,
  parseUser: req => {
    // custom user parsing logic
    return {
      email: req.user.email, // <-- here
      username: req.user.username,
      id: req.user.id
    }
  }
}).install()
```

I thought it was related to the user not being logged in (hence why `req.user` would be `undefined`), so I added this code, and now I'm not seeing the error.

```js
  if (!req.user) return {}
```

What's odd is that there was only one user on the site at the time (me), and I was logged in. Therefore, I'm not sure why `req.user` was `undefined`.

**Comment by mshwery** [comment]
I am seeing this same issue on `2.1.0` with an Express 4.x request object.

```js
function testErrorHandler (req, res) {
  Raven.captureException(err, { req: req })
}
```

Note I do have a `req.user` object but I have `parseUser: true` because the user object doesn't have the properties that Sentry looks for by default.

**Comment by benvinegar** [comment]
Re: "req parameter being discarded" – this is likely because the property (passed to Sentry) should be named `request`, not `req`. More in our [client SDK documentation](https://docs.sentry.io/clientdev/interfaces/http/).

This is probably a small oversight. Investigating this now.

@developius – that seems like a different issue.

**Comment by benvinegar** [comment]
Some more investigation: this almost certainly looks like `raven-node` was passing `req` as a top-level interface property. [Here](https://github.com/getsentry/sentry/blob/544a187885ebdab255b993c0cb97288ad9285a78/src/sentry/coreapi.py#L598) is where the error is triggered.

**Comment by benvinegar** [comment]
@ambar appears to have put together a fix in #339

**Comment by kamilogorek** [comment]
Already fixed. Closing.
